subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/javadoc/JavadocToolchainIntegrationTest.groovy

```diff
@@ -25,43 +25,48 @@ import spock.lang.Unroll
 
 class JavadocToolchainIntegrationTest extends AbstractIntegrationSpec {
 
+    def setup() {
+        executer
+            .withArgument("-Porg.gradle.java.installations.auto-detect=false")
+            .withArgument("-Porg.gradle.java.installations.auto-download=false")
+    }
+
     @Unroll
     @IgnoreIf({ AvailableJavaHomes.getJdk(JavaVersion.VERSION_11) == null })
-    def "can manually set javadoc tool via  #type toolchain on javadoc task #jdk"() {
+    def "can manually set javadoc tool via  #type toolchain on javadoc task #type : #jdk"() {
         buildFile << """
             plugins {
                 id 'java'
             }
 
-            javadoc {
+            // need to do as separate task as -version stop javadoc generation
+            task javadocVersionOutput(type: Javadoc) {
+                source = tasks.javadoc.source
+                options.jFlags("-version")
                 javadocTool = javaToolchains.javadocToolFor {
                     languageVersion = JavaLanguageVersion.of(${jdk.javaVersion.majorVersion})
                 }
             }
 
-            // need to do as separate task as -version stop javadoc generation
-            task javadocVersionOutput(type: Javadoc) {
-                options.jFlags("-version")
+            javadoc {
                 javadocTool = javaToolchains.javadocToolFor {
                     languageVersion = JavaLanguageVersion.of(${jdk.javaVersion.majorVersion})
                 }
+                dependsOn tasks.named("javadocVersionOutput")
```

Comment:
`dependsOn "javadocVersionOutput"` is enough, not a big deal, just choose the way you like.